engine/src/main/java/org/pentaho/di/trans/steps/pgbulkloader/PGBulkLoader.java

```diff
@@ -133,12 +136,40 @@ public String getCopyCommand( ) throws KettleException {
     return contents.toString();
   }
 
+  void checkClientEncoding() throws Exception {
+    Connection connection = data.db.getConnection();
+
+    Statement statement = connection.createStatement();
+
+    try {
+      try (ResultSet rs = statement.executeQuery("show client_encoding")) {
+        if (!rs.next()) {
+          throw new KettleException( "Cannot detect client encoding" );
+        }
+
+        if (rs.getMetaData().getColumnCount() != 1) {
+          throw new KettleException( "Cannot detect client encoding" );
+        }
+
+        String clientEncodingStr = rs.getString(1);
+        logBasic( "Detect client encoding: " + clientEncodingStr );
+        clientEncoding = Charset.forName(clientEncodingStr);
```

Comment:
Need a try/catch around this line to catch a few specific exceptions:
IllegalCharsetNameException - If the given charset name is illegal
IllegalArgumentException - If the given charsetName is null
UnsupportedCharsetException - If no support for the named charset is available in this instance of the Java virtual machine

In all three of these cases, charset will need to be set to null so that down below, it can be specified using the system default encoding.

Reply:
IllegalArgumentException is superclass of IllegalCharsetNameException and UnsupportedCharsetException

clientEncoding has system default charset as default value